README.md

```diff
@@ -29,20 +29,57 @@ describe('Blog page', () => {
 
 ## What
 
-The idea behind this library is to enable DOM integration tests on Next.js pages along with [data fetching][next-docs-data-fetching] and [routing][next-docs-routing].
+The idea behind this library is to reproduce as closer as possible the way Next.js works without spinning up servers, and render the output in a local JSDOM environment.
 
-The testing approach suggested here consists of manually mocking external API's dependencies and get the component instance matching a given route.
+In order to provide a valuable testing experience `next-page-tester` replicates the **rendering flow of an actual `next.js` application**:
 
-Next page tester will take care of:
+1. **fetch data** for a given route
+2. **render** the server-side rendered result to JSDOM as plain html (including `head` element)
+3. **mount/hydrate** the client application to the previously rendered html
 
+The mounted application is **interactive** and can be tested with any DOM testing library (like [`@testing-library/dom`](https://testing-library.com/docs/dom-testing-library/intro)).
+
+`next-page-tester` will take care of:
+
+- load and execute modules in the **expected browser or server environments**
 - **resolving** provided **routes** into matching page component
 - calling **Next.js data fetching methods** (`getServerSideProps`, `getInitialProps` or `getStaticProps`) if the case
 - set up a **mocked `next/router` provider** initialized with the expected values (to test `useRouter` and `withRouter`)
 - wrapping page with custom `_app` and `_document` components
-- **instantiating** page component with **expected page props**
-- emulating client side navigation via `Link`, `router.push`, `router.replace`
+- emulating **client side navigation** via `Link`, `router.push`, `router.replace`
 - handling pages' `redirect` returns
 
+## API
+
+### getPage
+
+`getPage` accepts an [option object](/#options) and returns 4 values:
+
+```ts
+function getPage(options: Options): Promise<
+  renderHTML: () => HTMLElement<NextRoot>
+  render: () => HTMLElement<NextRoot>
+  html: string
+  page: React.ReactElement<AppElement>
+>;
+```
+
+#### `renderHTML()`
+
+Inject server side rendering output to the DOM. Returns `#__next` root element element.
+
+#### `render()`
+
+Calls `renderHTML()` and mounts/hydrate the the application on `#__next` root element. Returns `#__next` root element element.
```

Comment:
Lets clarify when user would want to use each option. If I understand it correctly, `render` should be used in most cases, but `renderHTML` could be used if users want more control and do the `hydration` themselves?

Reply:
Yep `hydration` themselves but especially test what a search engine would see. 👍